numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -1494,17 +1494,24 @@ PyArray_Correlate(PyObject *op1, PyObject *op2, int mode)
 
 
 static PyObject *
-array_putmask(PyObject *NPY_UNUSED(module), PyObject *args, PyObject *kwds)
+array_putmask(PyObject *NPY_UNUSED(module), PyObject *const *args, Py_ssize_t len_args,
+                             PyObject *kwnames)
 {
     PyObject *mask, *values;
     PyObject *array;
 
-    static char *kwlist[] = {"arr", "mask", "values", NULL};
-
-    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O!OO:putmask", kwlist,
-                &PyArray_Type, &array, &mask, &values)) {
+    NPY_PREPARE_ARGPARSER;
+    if (npy_parse_arguments("putmask", args, len_args, kwnames,
+            "", NULL, &array,
+            "", NULL, &mask,
+            "", NULL, &values,
```

Comment:
This takes kwargs `mask` and `value` (test would be nice to have).  It looks like we accidentally deprecated the array being passed as a keyword argument which makes me happy :).

The python wrapper uses `a` while here we have `arr` so they disagree.  Could add the `, /,` to the python wrapper and `pyi` now that we can.